emspay.php

```diff
@@ -338,6 +341,12 @@ function ginger_filter_gateway_by_currency($gateways) {
         } else {
             wc_clear_notices();
         }
+        $ems_payment_methods = false; //flag that says gateways are contain ems methods or not
+        foreach ($gateways as $key => $gateway)
+        {
+            if (strstr($gateway->id,'emspay')) $ems_payment_methods = true;
```

Comment:
If keep foreach, I would add a break; after $ems_payment_methods = true; so as not to iterate over the entire array.
But I like this option:
if (! (int) preg_grep('/emspay/', array_keys($gateways))) return $gateways

Reply:
I will change to this option: if (! (int) preg_grep('/emspay/', array_keys($gateways))) return $gateways 
I like it more than foreach👍